src/components/forms/AddressInput/index.tsx

```diff
@@ -38,51 +50,88 @@ const AddressInput = ({
-            logError(Errors._101, err.message)
-          }
-        } else {
-          // A regular address hash
-          let checkedAddress = address
-          // Automatically checksum valid (either already checksummed, or lowercase addresses)
-          if (isValidAddress(address)) {
+}: AddressInputProps): React.ReactElement => {
+  const showNetworkPrefix = useSelector(showShortNameSelector)
+  const [prefixedAddress, setPrefixedAddress] = React.useState(() => addNetworkPrefix(value, showNetworkPrefix))
+  const [prefixedError, setPrefixedError] = React.useState(false)
+
+  return (
+    <>
+      <Field
+        className={className}
+        component={TextField as any}
+        defaultValue={defaultValue}
+        disabled={disabled}
+        inputAdornment={inputAdornment}
+        name={name}
+        inputProps={{
+          value: showNetworkPrefix ? prefixedAddress : value,
+          onChange: (e) => {
+            const value = e.target.value
+            setPrefixedAddress(value)
+
+            const hasPrefixedError = !!checkNetworkPrefix(value)
+
+            setPrefixedError(hasPrefixedError)
+
+            fieldMutator(value)
+          },
+          'data-testid': testId,
+        }}
+        placeholder={placeholder}
+        testId={testId}
+        text={text}
+        type="text"
+        spellCheck={false}
+        validate={composeValidators(required, checkNetworkPrefix, mustBeEthereumAddress, ...validators)}
+      />
+      <OnChange name={name}>
+        {async (newValue: string) => {
+          const trimmedValue = trimSpaces(newValue)
+
+          const address = prefixedError ? trimmedValue : getAddressWithoutNetworkPrefix(trimmedValue)
+
+          // A crypto domain name
+          if (isValidEnsName(address) || isValidCryptoDomainName(address)) {
             try {
-              checkedAddress = checksumAddress(address)
+              const resolverAddr = await getAddressFromDomain(address)
+              const formattedAddress = checksumAddress(resolverAddr)
+              fieldMutator(formattedAddress)
+              setPrefixedAddress(addNetworkPrefix(formattedAddress, showNetworkPrefix))
             } catch (err) {
-              // ignore
+              logError(Errors._101, err.message)
             }
+          } else {
+            // A regular address hash
+            let checkedAddress = address
+            // Automatically checksum valid (either already checksummed, or lowercase addresses)
+            if (isValidAddress(address)) {
+              try {
+                checkedAddress = checksumAddress(address)
+                const prefix = getNetworkPrefix(prefixedAddress)
+
+                // checksum the address in the input
+                if (prefix) {
+                  setPrefixedAddress(addNetworkPrefix(checkedAddress, showNetworkPrefix, prefix))
+                } else {
+                  // this is needed to avoid remove the : char and no prefix is present
```

Comment:
Could you rephrase this? Not clear what you mean.

Reply:
yes! sorry for that!
I refactored it to be more clear:
```javascript
                // checksum the address in the input
                checkedAddress = checksumAddress(address)
                const prefix = getNetworkPrefix(prefixedAddress)
                const isPrefixed = prefix || prefixedAddress.includes(':')

                // we set the correct network prefix in the state
                if (isPrefixed) {
                  setPrefixedAddress(addNetworkPrefix(checkedAddress, showNetworkPrefix, prefix))
                } else {
                  setPrefixedAddress(addNetworkPrefix(checkedAddress, showNetworkPrefix, getCurrentShortChainName()))
                }
 ```